amm/contracts/stable_pool/token_rate.rs

```diff
@@ -51,7 +51,8 @@ impl ExternalTokenRate {
         }
     }
 
-    pub fn get_rate_update(&mut self, current_block_no: u32) -> u128 {
+    pub fn get_rate_update(&mut self) -> u128 {
+        let current_block_no = ink::env::block_number::<DefaultEnvironment>();
```

Comment:
I know I initiated this change but is this for sure equivalent to `self.env().block_number()`? :stuck_out_tongue: